### PR TITLE
go stats: new availability v3 formula

### PR DESCRIPTION
**Pull request by pro-wh**
here's the implementation of the new availability score v3 for May

### CODE REVIEW

File: .changelog/2896.feature.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+go/stats: New availability v3 formula
```
Comment (kostko) [inline .changelog/2896.feature.md:1]:
```suggestion
go/extra/stats: New availability v3 formula
```

File: go/extra/stats/cmd/stats.go [hunk 2]
```diff
@@ -248,74 +282,84 @@ func getStats(ctx context.Context, consensus consensusAPI.ClientBackend, registr
...
-				logger.Error("failed to query registry",
+		tmBlockMeta := getTmBlockMetaOrExit(ctx, consensus, height)
+
+		// Process the commit that is put on chain in this block.
+		if height > 1 {
+			// Commit is for previous height.
+			lastCommitHeight := height - 1
+
+			// Get validators.
+			// Hypothesis: this gets the validator set with priorities after they're incremented to get the round 0
```
Comment (kostko) [inline go/extra/stats/cmd/stats.go:293]: I assume you checked that this is actually the case?

File: go/extra/stats/cmd/stats.go [hunk 3]
```diff
@@ -248,74 +282,84 @@ func getStats(ctx context.Context, consensus consensusAPI.ClientBackend, registr
...
 
-				// Add proposal.
-				if err := stats.addNodeProposal(previousProposerAddr); err != nil {
-					logger.Error("failure adding proposal",
-						"err", err,
-					)
-					os.Exit(1)
+			// Go over all signatures for a block.
+			for _, sig := range tmBlockMeta.LastCommit.Signatures {
+				if sig.Absent() {
```
Comment (kostko) [inline go/extra/stats/cmd/stats.go:320]: While we're here, what about signatures for nil? Are those recorded too (I believe with the Tendermint upgrade they are) and should they be treated as "not signing"?

Comment (pro-wh) [reply go/extra/stats/cmd/stats.go:320]: don't know

File: go/extra/stats/cmd/stats.go (hunk 2 above)
Comment (pro-wh) [reply go/extra/stats/cmd/stats.go:293]: I get the gist of it now

File: go/extra/stats/cmd/stats.go (hunk 3 above)
Comment (peterjgilbert) [reply go/extra/stats/cmd/stats.go:320]: If you voted for `nil` in a round that succeeded, you failed to sign the block that was committed. Is this any more useful that not signing?

Comment (pro-wh) [reply go/extra/stats/cmd/stats.go:320]: it's an online conscientious objection

Comment (pro-wh) [reply go/extra/stats/cmd/stats.go:321]: I'll make it not count. only popular answer wins ):

**Pull request merged**